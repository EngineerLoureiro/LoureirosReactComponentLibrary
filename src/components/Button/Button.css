.btn {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 6px;
  flex-grow: 1;
  padding: 0.5rem 1.25rem;
  border-radius: 4px;
  box-shadow: 0px 1px 2px 0 rgb(0 0 0 / 0.06), 0px 1px 3px 0 rgb(0 0 0 / 0.1);
  font-family: "Noto Sans";
  white-space: nowrap;
}

.btn:focus {
  box-shadow: 0px 0px 0px 4px
    rgb(0.2666666805744171 0.2980392277240753 0.9058823585510254 / 0.12);
}
.btn:disabled {
  background-color: #f5f5f5;
  color: #a3a3a3;
}

.btn-primary {
  background-color: #4338ca;
  box-shadow: 0px 1px 2px 0 rgb(0 0 0 / 0.06), 0px 1px 3px 0 rgb(0 0 0 / 0.1);
  color: #ffffff;
}

.btn-primary:hover,
.btn-primary:focus {
  background-color: #3730a3;
}

.btn-secondary {
  background-color: #ffffff;
  border-style: solid;
  border-width: 0.5px;
  border-color: #e5e5e5;
  box-shadow: 0px 1px 2px 0 rgb(0 0 0 / 0.06), 0px 1px 3px 0 rgb(0 0 0 / 0.1);
  color: #171717;
}

.btn-secondary:hover,
.btn-secondary:focus {
  background-color: #fafafa;
  border-width: 1px;
}
