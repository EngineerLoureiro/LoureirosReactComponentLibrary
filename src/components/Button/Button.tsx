import classNames from "classnames";
import "./Button.css";

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  label: string;
  backgroundColor?: string;
  textColor?: string;
  variant?: "primary" | "secondary";
}

export function Button({
  label,
  className,
  backgroundColor,
  textColor,
  style,
  onClick,
  variant,
  disabled,
  ...rest
}: ButtonProps) {
  return (
    <button
      className={classNames("btn", variant && `btn-${variant}`, className)}
      style={{ backgroundColor, color: textColor, ...style }}
      onClick={onClick}
      disabled={disabled}
      {...rest}
    >
      {label}
    </button>
  );
}
