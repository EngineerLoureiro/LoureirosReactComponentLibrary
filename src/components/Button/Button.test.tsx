import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./Button";

describe("Button", () => {
  it("renders with the correct label", () => {
    render(<Button label="Click Me" />);
    expect(screen.getByRole("button")).toHaveTextContent("Click Me");
  });

  it("calls onClick when clicked", () => {
    const handleClick = jest.fn();
    render(<Button label="Click Me" onClick={handleClick} />);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).toHaveBeenCalled;
  });

  it("does not call onClick when disabled", () => {
    const handleClick = jest.fn();
    render(<Button label="Click Me" onClick={handleClick} disabled={true} />);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).toHaveBeenCalledTimes(0);
  });

  it("applies the correct variant class", () => {
    render(<Button label="Secondary" variant="secondary" />);
    expect(screen.getByRole("button")).toHaveClass("btn-secondary");
  });

  it("renders primary button snapshot", () => {
    const { asFragment } = render(<Button label="Save" variant="primary" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders secondary button snapshot", () => {
    const { asFragment } = render(<Button label="Save" variant="secondary" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders disabled button snapshot", () => {
    const { asFragment } = render(<Button label="Save" disabled={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
