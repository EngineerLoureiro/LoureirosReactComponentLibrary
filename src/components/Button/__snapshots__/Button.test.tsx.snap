// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders disabled button snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn"
    disabled=""
  >
    Save
  </button>
</DocumentFragment>
`;

exports[`Button renders primary button snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
  >
    Save
  </button>
</DocumentFragment>
`;

exports[`Button renders secondary button snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-secondary"
  >
    Save
  </button>
</DocumentFragment>
`;
